pkg/kubelet/dockertools/manager.go

```diff
@@ -1261,33 +1271,46 @@ func (dm *DockerManager) KillPod(pod *api.Pod, runningPod kubecontainer.Pod) err
 				return
 			}
 
+			killContainerResult := &SyncResult{Action: KillContainer, Target: container.Name}
 			err := dm.KillContainerInPod(container.ID, containerSpec, pod, "Need to kill pod.")
 			if err != nil {
+				killContainerResult.Fail(kubecontainer.ErrKillContainer, err.Error())
 				glog.Errorf("Failed to delete container: %v; Skipping pod %q", err, runningPod.ID)
 				errs <- err
 			}
+			containerResults <- killContainerResult
 		}(container)
 	}
 	wg.Wait()
+	close(containerResults)
```

Comment:
@Random-Liu It's not your fault, but we need to remove `wg.Add(1)` in to loop and use `wg.Add(len(runningPod.Containers)` prior to the loop, because otherwise it will make race detector angry for go 1.5. See https://groups.google.com/forum/#!topic/golang-nuts/q7gmFZEF618

cc @yujuhong